master/internal/resourcemanagers/agent/agent.go

```diff
@@ -310,6 +310,7 @@ func (a *agent) handleIncomingWSMessage(ctx *actor.Context, msg aproto.MasterMes
 	switch {
 	case msg.AgentStarted != nil:
 		telemetry.ReportAgentConnected(ctx.Self().System(), a.uuid, msg.AgentStarted.Devices)
+		prom.AddAgentAsTarget(ctx, a.uuid.String(), a.address, a.resourcePoolName)
```

Comment:
might not matter (prom might not do anything) but i think this should be in `if !a.agentStartedInit {`

Reply:
this comment is no longer relevant since moving to HTTP SD configs for the targets